lib/schemaUtils.js

```diff
@@ -2977,9 +3004,14 @@ module.exports = {
   checkRequestHeaders: function (headers, transactionPathPrefix, schemaPathPrefix, schemaPath,
     components, options, schemaCache, callback) {
     let schemaHeaders = _.filter(schemaPath.parameters, (param) => { return param.in === 'header'; }),
-      // filter out headers which need explicit handling according to schema (other than parameters object)
+      // key name of headers which are added by security schemes
+      securityHeaders = _.map(this.getSecurityParams(_.get(components, 'components'), 'header'), 'name'),
+      // filter out headers for following cases
       reqHeaders = _.filter(headers, (header) => {
-        return !_.includes(IMPLICIT_HEADERS, _.toLower(_.get(header, 'key')));
+        // 1. which need explicit handling according to schema (other than parameters object)
+        // 2. which are added by security schemes
+        return !_.includes(IMPLICIT_HEADERS, _.toLower(_.get(header, 'key'))) &&
```

Comment:
shouldn't IMPLICIT_HEADERS also include Authorization?

Reply:
Oh yes, we discussed this also OpenAPI suggests this three definition to be ignored if present in param as header.
`Content-Type`, `Authorization` and `Content-Type`. So adding them.
see https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#fixed-fields-10